speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/confscreens/ThresholdConfig.cpp

```diff
@@ -0,0 +1,202 @@
+    char* val = GfuiEditboxGetString(s_scrHandle, m_steerThresholdControl);
+    char* endptr;
+    m_steerVal = strtof(val, &endptr);
+    if (*endptr != '\0')
+        std::cerr << "Could not convert " << val << " to long and leftover string is: " << endptr << std::endl;
+
+    if (m_steerVal > 1.0f)
+        m_steerVal = 1.0f;
+    else if (m_steerVal < 0.0f)
+        m_steerVal = 0.0f;
+
+    // Write the new (clamped) value to the text box.
+    char buf[32];
+    sprintf(buf, "%g", m_steerVal);
+    GfuiEditboxSetString(s_scrHandle, m_steerThresholdControl, buf);
+}
+
+/// @brief Read threshold config from local xml file
+static void ReadThresholdConfig()
+{
+    char buf[1024];
+
+    // Threshold interface.
+    sprintf(buf, "%s%s", GfLocalDir(), SND_PARAM_FILE);
+    void* paramHandle = GfParmReadFile(buf, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+
+    // Accel threshold
+    m_accelVal = GfParmGetNum(paramHandle, SND_SCT_THRESHOLD, SND_ATT_ACCEL, "%", STANDARD_THRESHOLD_ACCEL);
+    if (m_accelVal > 1.0f)
+    {
+        m_accelVal = 1.0f;
+    }
+    else if (m_accelVal < 0.0f)
+    {
+        m_accelVal = 0.0f;
+    }
+    sprintf(buf, "%g", m_accelVal);
+    GfuiEditboxSetString(s_scrHandle, m_accelThresholdControl, buf);
+
+    // Brake threshold
+    m_brakeVal = GfParmGetNum(paramHandle, SND_SCT_THRESHOLD, SND_ATT_BRAKE, "%", STANDARD_THRESHOLD_BRAKE);
+    if (m_brakeVal > 1.0f)
+    {
+        m_brakeVal = 1.0f;
+    }
+    else if (m_brakeVal < 0.0f)
+    {
+        m_brakeVal = 0.0f;
+    }
+    sprintf(buf, "%g", m_brakeVal);
+    GfuiEditboxSetString(s_scrHandle, m_brakeThresholdControl, buf);
+
+    // Steer threshold
+    m_steerVal = GfParmGetNum(paramHandle, SND_SCT_THRESHOLD, SND_ATT_STEER, "%", STANDARD_THRESHOLD_STEER);
+    if (m_steerVal > 1.0f)
+    {
+        m_steerVal = 1.0f;
+    }
+    else if (m_steerVal < 0.0f)
+    {
+        m_steerVal = 0.0f;
+    }
+    sprintf(buf, "%g", m_steerVal);
+    GfuiEditboxSetString(s_scrHandle, m_steerThresholdControl, buf);
+
+    GfParmReleaseHandle(paramHandle);
+}
+
+/// @brief Save the chosen values in the corresponding parameter file.
+static void SaveThresholdOptions(void*)
+{
+    // Force current edit to lose focus (if one has it) and update associated variable.
+    GfuiUnSelectCurrent();
+
+    char buf[1024];
+    sprintf(buf, "%s%s", GfLocalDir(), SND_PARAM_FILE);
+    void* paramHandle = GfParmReadFile(buf, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+    GfParmSetNum(paramHandle, SND_SCT_THRESHOLD, SND_ATT_ACCEL, "%", m_accelVal);
+    GfParmSetNum(paramHandle, SND_SCT_THRESHOLD, SND_ATT_BRAKE, "%", m_brakeVal);
+    GfParmSetNum(paramHandle, SND_SCT_THRESHOLD, SND_ATT_STEER, "%", m_steerVal);
+
+    GfParmWriteFile(nullptr, paramHandle, "Threshhold");
+    GfParmReleaseHandle(paramHandle);
+
+    // Update the Thresholds in the Mediator
+    SMediator::GetInstance()->SetThresholdSettings(buf);
```

Comment:
instead of setting the buf, just pass the `m_decisionThresholds` struct local from this file and set that as settings.